### PR TITLE
New guide: Cluster Formation

### PR DESCRIPTION
**Pull request by michaelklishin**
Which documents automatic cluster formation procedure and
the pluggable peer discovery mechanisms introduced in 3.7.0.

Part of https://github.com/rabbitmq/rabbitmq-server/issues/988 as well as https://github.com/rabbitmq/rabbitmq-server/issues/486.

### CODE REVIEW

File: site/autoclustering.xml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        cluster of the first reachable peer.
+      </p>
+
+      <p>
+        If peer discovery isn't configured, or it fails, or no peers are reachable,
+        the node will initialise from scratch and proceed as a standalone node.
+      </p>
+
+      <p>
+        Nodes that previously were cluster members will try to contact
```
Comment (acogoluegnes) [inline site/autoclustering.xml]: This sentence needs rewording.

File: site/autoclustering.xml [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+        </pre>
+      </p>
+    </doc:section>
+
+    <doc:section name="peer_discovery_dns">
+      <doc:heading>DNS Peer Discovery Backend</doc:heading>
+
+      <p>
+        Another build-in peer discovery mechanism as of RabbitMQ 3.7.0 is DNS-based.
```
Comment (acogoluegnes) [inline site/autoclustering.xml]: build-in => built-in

File: site/autoclustering.xml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+autocluster.classic_config.nodes.1 = rabbit@hostname1.eng.example.local
+autocluster.classic_config.nodes.2 = rabbit@hostname2.eng.example.local
+        </pre>
+      </p>
+      <p>
+        The following example in the classic config format. The 2nd
+        member of the <code>rabbit.cluster_nodes</code> tuple is the
+        node type to use for the current node. In the vast majority of
+        cases all nodes should be <code>disc</code>nodes.
```
Comment (dcorbacho) [inline site/autoclustering.xml]: Missing space between `disc` and `nodes`

File: site/autoclustering.xml [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+        <pre class="sourcecode">
+autocluster.peer_discovery_backend = rabbit_peer_discovery_classic_config
+
+autocluster.classic_config.nodes.1 = rabbit@hostname1.eng.example.local
+autocluster.classic_config.nodes.2 = rabbit@hostname2.eng.example.local
+        </pre>
+      </p>
+      <p>
+        The following example in the classic config format. The 2nd
```
Comment (dcorbacho) [inline site/autoclustering.xml]: Missing the verb on this sentence. Maybe `uses the classic config format`?

File: site/autoclustering.xml [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      </p>
+      <p>
+        The discovery mechanism to use is specified in the <a
+        href="/configure.html">config file</a>, as are
+        mechanism-specific settings, for example, discovery service
+        hostnames, credentials, and so on.
+        <code>autocluster.peer_discovery_backend</code> is the key
+        that controls what discovery module (implementation) is used.
+        The module has to implement the <a
+        href="https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_peer_discovery.erl">rabbit_peer_discovery</a>
```
Comment (dcorbacho) [inline site/autoclustering.xml]: It should be `rabbit_peer_discovery_backend` at https://github.com/rabbitmq/rabbitmq-common/blob/master/src/rabbit_peer_discovery_backend.erl

**Comment by michaelklishin** [comment]
Should be corrected now.

File: site/autoclustering.xml [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+        cluster of the first reachable peer.
+      </p>
+
+      <p>
+        If peer discovery isn't configured, or it fails, or no peers are reachable,
+        the node will initialise from scratch and proceed as a standalone node.
+      </p>
+
+      <p>
+        If a node node previously was a cluster member, it will try to contact
```
Comment (acogoluegnes) [inline site/autoclustering.xml]: "node node" => "node"

**Pull request merged**